src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageBus.cs

```diff
@@ -338,11 +327,34 @@ public override void Dispose()
     {
         base.Dispose();
 
+        if (_isDisposed)
+            return;
+
+        _isDisposed = true;
+
         if (_factory != null)
             _factory.AutomaticRecoveryEnabled = false;
 
         ClosePublisherConnection();
         CloseSubscriberConnection();
+        GC.SuppressFinalize(this);
+    }
+
+    public async ValueTask DisposeAsync()
```

Comment:
Followed this pattern https://alistairevans.co.uk/2019/10/24/net-asynchronous-disposal-tips-for-implementing-iasyncdisposable-on-your-own-types/ as well as what was happening in the rabbit client